source/slang/slang-compiler-tu.cpp

```diff
@@ -0,0 +1,79 @@
+// slang-compiler-tu.cpp: Compiles translation units to target language
+// and emit precompiled blobs into IR
+
+#include "../core/slang-basic.h"
+#include "slang-compiler.h"
+#include "slang-ir-insts.h"
+
+namespace Slang
+{
+    SLANG_NO_THROW SlangResult SLANG_MCALL Module::precompileForTargets(
+        DiagnosticSink* sink,
+        EndToEndCompileRequest* endToEndReq,
```

Comment:
why do we need EndToEndCompileRequest here? It shall only be needed by dxc and nothing else.

Reply:
You can get the linkage from the module itself, and the astbuilder from the linkage.

Reply:
I will address this in a follow-on change that makes this feature work with the Slang API, if that's okay